### ISSUE TITLE
Do not rely on Firebase for auth

### ISSUE THREAD

**Issue by michaelcpuckett** [michaelcpuckett/ap-groups#9]
After sharing widely via Feditips, some people have voiced concern about relying on Firebase.

**Comment by Andre601** [comment]
What are the main concerns actually?

**Comment by broukema** [comment]
The whole idea of the Fediverse is community run conversations without providing users - or moderators - as raw materials ("ordinary people are the product, not the client") to feed into GAFAM [1] for selling to advertisers [2]. Firebase appears to be run on Google/Alphabet servers. Using it for Fediverse groups defeats the point of the Fediverse.

Nuances: 
* A lot of Fediverse conversation is technically public, but it's equally public to everyone with internet access rather than just to one corporation, and there is a variety of opinions about the degree to which Fediverse conversations or meta-data should be propagated further. 
* Github itself is now controlled by GAFAM with attempts to get people to download repositorities using 'gh' instead of 'git' under the EEE (Embrace, Extend, Extinguish) policy of M; people remaining on Github are generally aware of this and waiting for further developments of Forgefed [3].

[1] https://en.wikipedia.org/wiki/GAFAM
[2] https://en.wikipedia.org/wiki/surveillance_capitalism
[3] https://forgefed.org

**Comment by michaelcpuckett** [comment]
Yes I think the concern is google. The site is hosted on DigitalOcean but authentication is handled by Firebase.

**Comment by michaelcpuckett** [comment]
I would like to implement an alternate authentication method, probably: Passport.js + existing mongo database

**Comment by Andre601** [comment]
Yeah.... security doesn't seem *that* safe with this setup...

**Comment by goodevilgenius** [comment]
And by the way, the concern with Firebase isn't necessarily *just* that it's Google, but also that it's a proprietary, closed-source service.

If auth was handled directly, then I can see the code and know what it's doing with my password (presumably, hashing it and adding the hash to the database). But, with Firebase, I have no way of knowing exactly what is happening to my email and password.

Handing auth off to Firebase makes this service not fully open-source, because of that dependancy.

**Comment by goodevilgenius** [comment]
@Andre601 what seems unsafe about that setup? Are you worried that someone could spoof a Fediverse account?

ActivityPub handles that with encryption keys and signatures: https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/#http-signatures

**Comment by Andre601** [comment]
> @Andre601 what seems unsafe about that setup? Are you worried that someone could spoof a Fediverse account?
> 
> ActivityPub handles that with encryption keys and signatures: https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/#http-signatures

Well, for once is Mastodon, which a lot of people use, not end-to-end encrypted in DMs, so there's no guarantee that someone couldn't find info and alike.
